### PR TITLE
Add catalog for build mattermost app in rancher

### PR DESCRIPTION
**Pull request by h3d-thanhpv**
This catalog enables deployment of a Mattermost server in production configuration using Docker and Rancher Cattle.

Notes:
 - The default Mattermost edition for this repo is Team Edition

### CODE REVIEW

File: templates/mattermost/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+version: '2'
+services:
+  mattermost:
+    image: thanhpv102/mattermost-team
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Don't use generic docker image tag, implicit neither explicit, they are not reproducible. Please use concrete docker image version. 
```
-    image: thanhpv102/mattermost-team
+    image: thanhpv102/mattermost-team:4.3.0
```

File: templates/mattermost/0/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: 2
+catalog:
+  name: Mattermost
+  version: v0.1.0
```
Comment (rawmind0) [inline templates/mattermost/0/rancher-compose.yml]: What do you think to match package version with Mattermost version??
```
-  version: v0.1.0
+  version: v4.3.0
```

File: templates/mattermost/config.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+name: Mattermost
+description: |
+  Open source, private cloud Slack-alternative, Workplace messaging for web, PCs and phones
+version: v0.1.0
```
Comment (rawmind0) [inline templates/mattermost/config.yml]: What do you think to match package version with Mattermost version??
```
-version: v0.1.0
+version: v4.3.0
```

File: templates/mattermost/0/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  description: |
+    Open source, private cloud Slack-alternative, Workplace messaging for web, PCs and phones
+  minimum_rancher_version: v0.51.0
+  maintainer: "Thanh Phan <thanhpv.102@gmail.com>"
+  uuid: mattermost-0
```
Comment (rawmind0) [inline templates/mattermost/0/rancher-compose.yml]: Please, add question for postgresql db name, user and password ...
```
    - variable: mm_dbname
      description: "Mattermost database name"
      label: "MM_DBNAME"
      required: true
      default: "mattermost"
      type: "string"
    - variable: mm_username
      description: "Mattermost db user name"
      label: "MM_USERNAME"
      required: true
      default: "mmpostgres"
      type: "string"
    - variable: mm_password
      description: "Mattermost db password"
      label: "MM_PASSWORD"
      required: true
      type: "password"
      default: "mmpassword"
```

File: templates/mattermost/0/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    environment:
+      - MM_DBNAME=mattermost
+      - MM_PASSWORD=mmpassword
+      - MM_USERNAME=mmpostgres
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Please, change hardcoded values with rancher-compose questions values.
```
-      - MM_DBNAME=mattermost
-      - MM_PASSWORD=mmpassword
-      - MM_USERNAME=mmpostgres
+      - MM_DBNAME=${mm_dbname}
+      - MM_PASSWORD=${mm_password}
+      - MM_USERNAME=${mm_username}
```

File: templates/mattermost/0/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      - DB_HOST=mattermostdb
+    volumes:
+      - mattermost-data:/mattermost/data
+  mattermostdb:
+    image: mattermost/mattermost-prod-db:4.3.0
+    environment:
+      - POSTGRES_USER=mmpostgres
+      - POSTGRES_PASSWORD=mmpassword
+      - POSTGRES_DB=mattermost
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Please, change hardcoded values with rancher-compose questions values.
```
-      - POSTGRES_USER=mmpostgres
-      - POSTGRES_PASSWORD=mmpassword
-      - POSTGRES_DB=mattermost
+      - POSTGRES_USER=${mm_username}
+      - POSTGRES_PASSWORD=${mm_password}
+      - POSTGRES_DB=${mm_dbname}
```

File: templates/mattermost/0/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    volumes:
+      - mattermost-db:/var/lib/postgresql/data
+volumes:
+  mattermost-data:
+  mattermost-db:
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: You are not publishing any port to consume the service, is it right??

**Comment by rawmind0** [comment]
Hi @thanhpv-102 , thanks so much for the PR and contribute.

Please, take a look at the changes requested.

**Comment by thanhpv-102** [comment]
Thanks for your reviews. Now I implement it.

File: templates/mattermost/0/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+version: '2'
+services:
+  mattermost:
+    image: thanhpv102/mattermost-team:4.3.0
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: This docker image doesn't exist. may be 4.3.1??
```
-    image: thanhpv102/mattermost-team:4.3.0
+    image: thanhpv102/mattermost-team:4.3.1
```

File: templates/mattermost/0/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      - mattermost-data:/mattermost/data
+  mattermostdb:
+    image: mattermost/mattermost-prod-db:4.3.0
+    environment:
+      - POSTGRES_USER=${mm_username}
+      - POSTGRES_PASSWORD=${mm_password}
+      - POSTGRES_DB=${mm_dbname}
+    volumes:
+      - mattermost-db:/var/lib/postgresql/data
+  letschatlb:
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Not same name for lb service at rancher compose
```
-  letschatlb:
+  mattermostlb:
```

File: templates/mattermost/0/docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    image: rancher/lb-service-haproxy:v0.7.5
+    ports:
+      - 80
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Specify internal and external port
```
-      - 80
+      - 80:80
```

File: templates/mattermost/0/rancher-compose.yml [hunk 11]
```diff
@@ -0,0 +1,56 @@
+version: 2
+catalog:
+  name: Mattermost
+  version: v4.3.0
```
Comment (rawmind0) [inline templates/mattermost/0/rancher-compose.yml]: May be v4.3.1??
```
-  version: v4.3.0
+  version: v4.3.1
```

File: templates/mattermost/1/docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+      - mattermost-data:/mattermost/data
+  mattermostdb:
+    image: mattermost/mattermost-prod-db:4.5.0
+    environment:
+      - POSTGRES_USER=${mm_username}
+      - POSTGRES_PASSWORD=${mm_password}
+      - POSTGRES_DB=${mm_dbname}
+    volumes:
+      - mattermost-db:/var/lib/postgresql/data
+  letschatlb:
```
Comment (rawmind0) [inline templates/mattermost/1/docker-compose.yml]: Not same name for lb service at rancher compose
```
-  letschatlb:
+  mattermostlb:
```

File: templates/mattermost/1/docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    image: rancher/lb-service-haproxy:v0.7.5
+    ports:
+      - 80
```
Comment (rawmind0) [inline templates/mattermost/1/docker-compose.yml]: Specify internal and external port
```
-      - 80
+      - 80:80
```

File: templates/mattermost/0/docker-compose.yml [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  mattermostdb:
+    image: mattermost/mattermost-prod-db:4.3.0
+    environment:
+      - POSTGRES_USER=${mm_username}
+      - POSTGRES_PASSWORD=${mm_password}
+      - POSTGRES_DB=${mm_dbname}
+    volumes:
+      - mattermost-db:/var/lib/postgresql/data
+  letschatlb:
+    image: rancher/lb-service-haproxy:v0.7.5
```
Comment (rawmind0) [inline templates/mattermost/0/docker-compose.yml]: Use latest lb service image.
```
-    image: rancher/lb-service-haproxy:v0.7.5
+    image: rancher/lb-service-haproxy:v0.7.9
```

File: templates/mattermost/1/docker-compose.yml [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+  mattermostdb:
+    image: mattermost/mattermost-prod-db:4.5.0
+    environment:
+      - POSTGRES_USER=${mm_username}
+      - POSTGRES_PASSWORD=${mm_password}
+      - POSTGRES_DB=${mm_dbname}
+    volumes:
+      - mattermost-db:/var/lib/postgresql/data
+  letschatlb:
+    image: rancher/lb-service-haproxy:v0.7.5
```
Comment (rawmind0) [inline templates/mattermost/1/docker-compose.yml]: Use latest lb service image.
```
-    image: rancher/lb-service-haproxy:v0.7.5
+    image: rancher/lb-service-haproxy:v0.7.9
```

**Comment by rawmind0** [comment]
Hi @thanhpv-102, some errors at the package, please test before commit.

Take a look at the changes requested.

**Pull request closed without merging**